### DISCUSSION TITLE
Type 'Signal<string>' is not assignable to type '{ [SIGNAL]: unknown; }'.

### DISCUSSION THREAD

**Discussion by mark-norgate-ons** [ngrx/platform#4228] [Q&A]
When using `withComputed` and `signalStore`, I have suddenly run into an issue where it appears the type cannot be inferred. This has been working, and even with reverting each change I am struggling to pinpoint where it went wrong!

A minimal but sufficient fragment:
```
import { signalStore, withComputed, withState } from '@ngrx/signals';
import { computed } from '@angular/core';

export const CategoryListStore = signalStore(
  withState<{ relaunchStatus: string }>({
    relaunchStatus: 'submitted'
  }),
  withComputed(({ relaunchStatus }) => ({
    filters: computed(() => relaunchStatus()) // <=== error occurs here
  }))
);
```
It was more elaborate than this, of course, but the error is reproducible in this fragment.
> <html>TS2322: Type 'Signal&lt;string&gt;' is not assignable to type 'Signal&lt;unknown&gt;'.<br/>Type 'Signal&lt;string&gt;' is not assignable to type '{ [SIGNAL]: unknown; }'.

I am using Nx: "@nx/angular": "17.3.0"

"@angular/core": "17.1.1"
"@ngrx/signals": "^17.1.0"

**Comment by MeMeMax** [comment]
I got the exact same problem. I am already on Angular 19. It worked and suddenly I got this error. Did you find a solution @mark-norgate-ons

**Comment by MeMeMax** [comment]
Just found out the problem. I made a stackoverflow post: https://stackoverflow.com/questions/79592657/type-signalboolean-is-not-assignable-to-type-signalunknown
